docs/backend.openapi.yaml

```diff
@@ -113,6 +113,61 @@ paths:
           description: User successfully signed out
         "403":
           description: Access denied
+  
+  /chat/users/{username}/messages:
+    put:
+      summary: User sends a message
+      tags:
+        - chat
+      security:
+        - auth: []
+      parameters:
+        - $ref: "#/components/parameters/username"
+      responses:
+        "200":
+          description: Message was successfully handled
+        "403":
+          description: Access denied
+
+  /chat/messages:
+    get:
+      summary: Last 100 messages in the chat
+      tags:
+        - chat
+      security:
+        - auth: []
+      responses:
+        "200":
+          description: List of last messages
+          content:
+            application/json:
+              schema:
+                type: array
+                items: 
+                  $ref: "#/components/schemas/Message" 
+        "403": 
+          description: Access denied
```

Comment:
There should be only 1 endpoint for retrieving messages:

1. If client doesn't send "cursor" in the request then server returns first portion of the messages and returns a "cursor" that points to the beginning of the following portion.

1. If client includes a "cursor" then server returns messages portion that starts from it.

The cursor is typically a `String` that is treated by a client as opaque; It means that client never makes any assumptions about the value, only passes it back to a server as is.